src/main/kotlin/app/revanced/patches/shared/settings/preference/impl/PreferenceScreen.kt

```diff
@@ -26,4 +26,15 @@ internal open class PreferenceScreen(
             for (childPreference in preferences)
                 this.appendChild(childPreference.serialize(ownerDocument, resourceCallback))
         }
+
+    fun addPreferences(vararg preferencesToAdd: BasePreference) {
+        val list: MutableList<BasePreference>
+        if (preferences is MutableList) {
+            list = preferences as MutableList<BasePreference>
+        } else {
+            list = preferences.toMutableList()
+            preferences = list
+        }
+        list.addAll(preferencesToAdd)
+    }
```

Comment:
That's an exciting solution, but I don't think it's pretty because this method arbitrarily exists in `PreferenceScreen` solely to solve the issue in this PR. `PreferenceScreen` has a property `preferences` where you can add preferences already. Currently, ReVanced has a different way of solving the problem that is being solved with this method. Create an intermediary "Seekbar" patch and make it implement "Closeable." Add an "addPreference" function to its companion object. The function collects added preferences. Now depend on the "Seekbar" patch everywhere where necessary. When all patches depend on it, the close function implemented from "Closeable" will be called. You can access the list of preferences previously added through the "addPreference" function and add them to SettingsPreference.PreferenceScreen.LAYOUT.